UI/platform-x11.cpp

```diff
@@ -256,3 +259,50 @@ bool SetDisplayAffinitySupported(void)
 // Not implemented yet
 void TaskbarOverlayInit() {}
 void TaskbarOverlaySetStatus(TaskbarOverlayStatus) {}
+
+bool HighContrastEnabled()
+{
+	QDBusReply<QVariant> reply;
+	QDBusMessage msgXdpSettingsVersion = QDBusMessage::createMethodCall(
+		"org.freedesktop.portal.Desktop",
+		"/org/freedesktop/portal/desktop",
+		"org.freedesktop.DBus.Properties", "Get");
+	msgXdpSettingsVersion << "org.freedesktop.portal.Settings"
+			      << "version";
+
+	reply = QDBusConnection::sessionBus().call(msgXdpSettingsVersion);
+
+	if (!reply.isValid()) {
+		blog(LOG_WARNING,
+		     "Get on org.freedesktop.portal.Setttings returned an invalid reply");
+		return false;
+	}
+
+	/* NOTE: org.freedesktop.portal.Settings got its constast settings after the ReadOne method.
+	 * So assumes that if ReadOne is not availlable, constrast isn't present availlable either.*/
```

Comment:
```suggestion
	 * So assumes that if ReadOne is not available, contrast isn't present available either. */
```

May need line-wrapped.